### PR TITLE
2 users1 room

### PR DESCRIPTION
**Pull request by Hannibal42**
See issue #69

### CODE REVIEW

File: tests/frontend/test_facilities.py [hunk 1]
```diff
@@ -51,3 +51,8 @@ def test_0080_building_level_rooms(self):
             "/facilities/buildings/{}/levels/{}/rooms/".format(
                 self.building.id, self.room.level),
             "facilities/rooms.html")
+
+    def test_0090_overcrowded_rooms(self):
```
Comment (lukasjuhrich) [inline tests/frontend/test_facilities.py]: Great that you provided a frontend test!

Minor detail, please remove the number from your new test case.
I know a lot of tests contain these, but including a number to enforce an execution order is pointless and unidiomatic.
Most of the tests have been left that way, but don't let that confuse you.

**Comment by Hannibal42** [comment]
Done.

**Comment by sebschrader** [comment]
Some rooms are allowed to be occupied by more than one person. Such rooms are excluded by the current system used at Wundtstraße as long as there were not more than two inhabitants.

File: web/blueprints/facilities/__init__.py [hunk 2]
```diff
@@ -40,6 +42,36 @@ def root():
...
     return render_template('facilities/site_overview.html')
 
+@bp.route('/overcrowded')
+@nav.navigate(u"Two users one room")
+def overcrowded():
+    return render_template("facilities/room_overcrowded.html")
+
+@bp.route('/overcrowded/json')
+def overcrowded_json():
+    query = session.session.query(User.room_id, func.array_agg(User.id), func.array_agg(User.name))\
```
Comment (lukasjuhrich) [inline web/blueprints/facilities/__init__.py]: Just querying for the Room ID is not sufficient, we need the full ORM object in order to display a useful string representation. However, something like `Room.q.join(User).group_by(Room.id).having(func.count(User.account_id) > 1).add_columns(func.array_agg(User.id)).all()
` does not work for a curious error („Unhashable type: List“)

**Comment by lukasjuhrich** [comment]
@Hannibal42 Do you have capacities to finish this PR?

Perhaps @sebschrader has some ideas how we could efficiently fetch all ORM user objects without doing something like `User.q.get(i)` in a for-loop over the last query's result.

**Comment by sebschrader** [comment]
The SQL query you suggested doesn't male sense. You can't group by some columns A and then access columns other columns than A without using aggregate functions on these columns.

I'll try to take a look at lit later. Probably a window function will solve the problem.

**Comment by Hannibal42** [comment]
I can have a look in the next few days, hopefully at saturday. But i first have to setup pycroft on my new system.

**Comment by sebschrader** [comment]
A window function is not necessary. The following self semi-join with the result of grouping rooms by the people that live three should do the trick.

```sql
SELECT r1.*
  FROM room r1 JOIN (
    SELECT r2.id
    FROM room r2
    JOIN "user" u
    ON u.room_id = r2.id
    GROUP BY r2.id
    HAVING COUNT(u.id) > 1
  ) r2
  ON r1.id = r2.id;
```

If I find the time, I try to translate the query to SQLAlchemy ORM.

**Comment by ibot3** [comment]
Implemented with https://github.com/agdsn/pycroft/pull/276.

**Pull request closed without merging**